bigtable/metrics.go

```diff
@@ -0,0 +1,401 @@
+			attribute.String(metricLabelKeyClientUID, clientUID),
+			attribute.String(metricLabelKeyClientName, clientName),
+		},
+		shutdown: func() {},
+	}
+
+	var meterProvider *sdkmetric.MeterProvider
+	if metricsProvider == nil {
+		// Create default meter provider
+		mpOptions, err := builtInMeterProviderOptions(project)
+		if err != nil {
+			return tracerFactory, err
+		}
+		meterProvider = sdkmetric.NewMeterProvider(mpOptions...)
+
+		tracerFactory.enabled = true
+		tracerFactory.shutdown = func() { meterProvider.Shutdown(ctx) }
+	} else {
+		switch metricsProvider.(type) {
+		case NoopMetricsProvider:
+			tracerFactory.enabled = false
+			return tracerFactory, nil
+		default:
+			tracerFactory.enabled = false
+			return tracerFactory, errors.New("unknown MetricsProvider type")
+		}
+	}
+
+	// Create meter and instruments
+	meter := meterProvider.Meter(builtInMetricsMeterName, metric.WithInstrumentationVersion(internal.Version))
+	err = tracerFactory.createInstruments(meter)
+	return tracerFactory, err
+}
+
+func builtInMeterProviderOptions(project string) ([]sdkmetric.Option, error) {
+	defaultExporter, err := newMonitoringExporter(context.Background(), project, exporterOpts...)
+	if err != nil {
+		return nil, err
+	}
+
+	return []sdkmetric.Option{sdkmetric.WithReader(
+		sdkmetric.NewPeriodicReader(
+			defaultExporter,
+			sdkmetric.WithInterval(defaultSamplePeriod),
+		),
+	)}, nil
+}
+
+func (tf *builtinMetricsTracerFactory) createInstruments(meter metric.Meter) error {
+	var err error
+
+	// Create operation_latencies
+	tf.operationLatencies, err = meter.Float64Histogram(
+		metricNameOperationLatencies,
+		metric.WithDescription("Total time until final operation success or failure, including retries and backoff."),
+		metric.WithUnit("ms"),
+		metric.WithExplicitBucketBoundaries(bucketBounds...),
+	)
+	if err != nil {
+		return err
+	}
+
+	// Create attempt_latencies
+	tf.attemptLatencies, err = meter.Float64Histogram(
+		metricNameAttemptLatencies,
+		metric.WithDescription("Client observed latency per RPC attempt."),
+		metric.WithUnit("ms"),
+		metric.WithExplicitBucketBoundaries(bucketBounds...),
+	)
+	if err != nil {
+		return err
+	}
+
+	// Create server_latencies
+	tf.serverLatencies, err = meter.Float64Histogram(
+		metricNameServerLatencies,
+		metric.WithDescription("The latency measured from the moment that the RPC entered the Google data center until the RPC was completed."),
+		metric.WithUnit("ms"),
+		metric.WithExplicitBucketBoundaries(bucketBounds...),
+	)
+	if err != nil {
+		return err
+	}
+
+	// Create retry_count
+	tf.retryCount, err = meter.Int64Counter(
+		metricNameRetryCount,
+		metric.WithDescription("The number of additional RPCs sent after the initial attempt."),
+	)
+	return err
+}
+
+// builtinMetricsTracer is created one per operation
+// It is used to store metric instruments, attribute values
+// and other data required to obtain and record them
+type builtinMetricsTracer struct {
```

Comment:
There are three different tracers in this file. Could you rename `builtinMetricsTracer` to something that differentiates it from `opTracer` and `attemptTracer`?